code/modules/mob/living/basic/pets/penguin.dm

```diff
@@ -76,9 +76,10 @@
 	remove_egg()
 
 /mob/living/basic/pet/penguin/proc/remove_egg()
-	if(carried_egg)
-		carried_egg.forceMove(get_turf(src))
-		UnregisterSignal(carried_egg, COMSIG_QDELETING)
+	if(!carried_egg)
```

Comment:
do `if(isnull(carried_egg))` instead

Reply:
i will but waht is the advantages isnull vs !